src/caffeine/caffeine.c

```diff
@@ -90,6 +90,15 @@ void caf_deallocate(mspace heap, void* mem)
   mspace_free(heap, mem);
 }
 
+// take address in a segment and convert to an address on given image
+void* caf_convert_base_addr(void* addr, int image, gex_TM_t initial_team)
+{
+   size_t offset = (char*)addr - (char*)gex_Segment_QueryAddr(mysegment);
```

Comment:
Shouldn't this be `ptrdiff_t`? And I always wonder, why `(char*)`?

Reply:
`ptrdiff_t` would also be correct here. This `offset` is guaranteed to be non-negative, so the signedness is irrelevant.

The `(char*)` is mandatory because pointer arithmetic on `void*` has unspecified behavior.

Reply:
I understand that you can't do pointer math on a raw `void*`, but why `char` specifically?

Reply:
Any single-byte type would work; `char` is guaranteed to be a single byte. 

I usually use `uint8_t` (or `std::byte` in C++) for clarity/documentation in such circumstances, but they mean the same thing.

Reply:
> I usually use `uint8_t` (or `std::byte` in C++) for clarity/documentation in such circumstances

Perhaps we should do so here too then. Or is it common to see something like `#define byte char`?

Reply:
>  is it common to see something like #define byte char?

It may be common, but I'd consider that a poor coding practice because: 
1) Never use a macro where a typedef would work, and 
2) `char` is "special" in a few other ways that IMO should not be obscured by macro/typedef.

I would not be opposed to `typedef uint8_t byte;` if we really want a shorthand, as long as such a declaration is never exported to Caffeine clients. I don't think we're planning to provide clients with any public header file in C for Caffeine, but I thought I'd mention it for completeness.

Reply:
I have used the suggested typedef.